realm/realm-library/src/main/cpp/io_realm_internal_Table.cpp

```diff
@@ -666,6 +666,25 @@ JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeSetLong(JNIEnv* env, j
     CATCH_STD()
 }
 
+JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeIncrementLong(JNIEnv* env, jclass, jlong nativeTablePtr,
+                                                                  jlong columnIndex, jlong rowIndex, jlong value)
+{
+    if (!TBL_AND_INDEX_AND_TYPE_VALID(env, TBL(nativeTablePtr), columnIndex, rowIndex, type_Int)) {
+        return;
+    }
+
+    Table* table = TBL(nativeTablePtr);
+    if (table->is_null(columnIndex, rowIndex)) {
+        ThrowException(env, IllegalArgument, "Cannot increment a MutableRealmInteger whose value is null. Set its value first.");
```

Comment:
we try to use the new way to throw java exception in JNI code.
just move line 676-680 to try block and then change this line to:
```
THROW_JAVA_EXCEPTION(env, JavaExceptionDef::IllegalArgument,
                         "Cannot increment a MutableRealmInteger whose value is null. Set its value first.");
```

You can search for `THROW_JAVA_EXCEPTION` to get some examples.